posts/2021-08-06-21009-beta.adoc

```diff
@@ -0,0 +1,177 @@
+# Do NOT change the categories section
+categories: blog
+author_picture: https://avatars3.githubusercontent.com/ryan-storey
+author_github: https://github.com/ryan-storey
+seo-title: Endpoint activation monitoring and more now available on Open Liberty 21.0.0.9-beta - OpenLiberty.io
+seo-description: Open Liberty 21.0.0.9-beta offers exciting new enhancements for Liberty embedders such as endpoint activation monitoring to determine when endpoints are available to use. 
+blog_description: Open Liberty 21.0.0.9-beta offers exciting new enhancements for Liberty embedders such as endpoint activation monitoring to determine when endpoints are available to use. 
+open-graph-image: https://openliberty.io/img/twitter_card.jpg
+---
+= Endpoint activation monitoring and more now available on Open Liberty 21.0.0.9-beta
+Ryan Storey <https://github.com/ryan-storey>
+:imagesdir: /
+:url-prefix:
+:url-about: /
+//Blank line here is necessary before starting the body of the post.
+
+
+Open Liberty 21.0.0.9-beta offers exciting new enhancements for Liberty embedders such as endpoint activation monitoring to determine when endpoints are available to use. 
+
+We have two beta packages for link:{url-about}[Open Liberty]:
+
+* <<allbeta, All Beta Features>>: a larger package that contains all Open Liberty beta features (including Jakarta EE 9 beta features) and GA features and functions.
+* <<jakarta, Jakarta EE 9 Beta Features>>: a lightweight package that contains only the Jakarta EE 9 features.
+
+This means that you can now try out our in-development Open Liberty features by just adding the relevant coordinates to your build tools.
+
+If you try either package, <<feedback, let us know what you think>>.
+[#allbeta]
+== All Beta Features package
+
+The All Beta Features package includes the following beta features:
+
+* <<method, New method to determine active endpoints>>
+* link:https://openliberty.io/blog/2021/07/12/logging-format-21008-beta.html#mp4_1[MicroProfile 4.1]
+
+
+[#method]
+=== New method to determine active endpoints
+
+The `ServerEndpointControlMBean` can currently be used to pause and resume endpoints, including `HTTP` and messaging ports. It can also determine if an endpoint is currently paused. However, there is no way to determine programmatically if an endpoint exists and is active. This update provides a new method on the `mbean`, `isActive`, that will determine if an endpoint exists, has started, and is not paused.
+A new method was added to the `ServerEndpointControlMBean`:
+
+`public boolean isActive(String targets);`
+
+The `targets` parameter is a comma separated list of endpoint names. 
+
+This mirrors the existing `isPaused` method. It will return true only if all of the endpoints listed exist, are started, and are not paused.
+
+[source, java]
+----
+// check the specified targets are active
+assertTrue("isActive should return true because all endpoints are active", mbean.isActive("defaultHttpEndpoint"));
```

Comment:
Do we usually include test code like this? My guess is that most people reading would be familiar with junit syntax, but it seems like it would be more straightforward to have an example like:

// Check if the defaultHttpEndpoint is active
boolean isEndpointActive = mbean.isActive("defaultHttpEndpoint");

Reply:
Changes made!